impress-org/givewp
# Refactor: general tweaks to adjust the core to the recent changes implemented in the Authorize.Net gateway addon

## Description



**This PR implements the following:**

- A new React component to be used on the Donor Dashboard when the user updates the payment method (credit card) for Authorize subscriptions. Now instead of sending the raw Credit Card data to the backend, we send the "data descriptor" and "data value" params generated by the [Accept.Js](https://developer.authorize.net/api/reference/features/acceptjs.html) library.

**Beyond the Authorize.Net gateway, the items below will benefit other gateways as well:**

- A way to handle update errors, on the donor dashboard, thrown by the backend during the update process.

- Changes the way that the form classic template adds the gateway field in the DOM, now it uses jQuery, and this way if any scripts are present with the fields, they also will be loaded.

- Adds a new donor email notification for donations with the "processing" status + notice on the "thank you" page informing the user that a final receipt will be sent to the email when the donation is completed.

## Affects



- Subscriptions created with Authorize.Net using the new gateway API and the Accept.js library
- Email Notifications
- Donor Dashboard
- Donation Receipt

## Visuals



![image](https://user-images.githubusercontent.com/16308864/205330585-51f02336-b6f3-482b-949f-15c33b4bf53d.png)

![image](https://user-images.githubusercontent.com/16308864/206241203-7fb954be-a34e-4457-9bfe-10f5393adf65.png)

![image](https://user-images.githubusercontent.com/16308864/210555968-2c86e133-af31-41d8-8059-34a52a0024de.png)

![image](https://user-images.githubusercontent.com/16308864/210555872-bf6980a4-78d3-46b9-8aa8-e18dabe65e7f.png)

![image](https://user-images.githubusercontent.com/16308864/211854320-5651d52b-7f66-4b78-afbd-a3e29b170841.png)

## Testing Instructions



First, pull the following PRs and setup the Authorize gateway:

https://github.com/impress-org/give-authorize-gateway/pull/111

https://github.com/impress-org/give-recurring/pull/1156

Second, create Authorize subscription using [a test credit card](https://developer.authorize.net/hello_world/testing_guide.html) and check if the thank you page has a notice just below the donation status and also if the "Donation Processing Receipt" notification was sent to the email. After that go to the donor dashboard and edit the credit card data to another one.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203580457557897
